Source/Core/DolphinQt2/Config/Graphics/GraphicsSlider.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 Dolphin Emulator Project
```

Comment:
I wouldn't call it a preference! It's an engineering claim.

I see a number of advantages to doing it with pure functions:
- Better readability: A `GraphicsButton` (for example) is a class that a reader probably isn't familiar with. When reading code using it, they'll have to context switch, go to another file, and figure out what it does. This is a lot harder than skimming code that uses generic, well-known `QPushButtons` and sets attributes/signals in-line.

- Better semantics: The `GraphicsChoice` class (for example) would be more explicitly named `QComboBoxBoundToAConfigInfo`, which is a complete description of what it does. However, class usually aren't named with past-tense verbs, because verbs are mainly supposed to be for functions. Semantically, this is a combo box that has had some function applied to it, binding it to a config info.

- Better possibility for DRY/coalescing logic: The code for each of these widgets looks essentially like:
  ```c++
  connect(widget, &Widget::ValueChanged, [](new_value) {
    Config::SetBaseOrCurrent(setting, InvolutoryTransform(new_value));
  })
  setValue(InvolutoryTransform(Config::Get(setting));
  if (Config::GetActiveLayerForConfig(setting) != Config::LayerType::Base)
  {
    QFont bf = font();
    bf.setBold(true);
    setFont(bf);
  }
  ```

  With the appropriate setup (not saying I want you to do this, it's probably too complex to be worthwhile right now), this could be shared across all the needed widget types.

  It also makes it more obvious when shared logic is missing, and where there are potential opportunities for extracting repeated code: Why doesn't `GraphicsSlider` set its font to bold, when the others do? Because a slider widget doesn't come attached to a description label or value label. This seems like a Qt oversight, and there might be other code that could be made simpler with a combination-slider-attached-to-its-own-labels (for example, the Audio pane, when sliders need to enabled/disabled on the fly).

- Better separation of controllers and views: In the classical architecture, all control/model logic is separated from the UI elements. Obviously we're not adhering to this 100%, but it's a good idea to keep them separated as much as possible. The fewer places that conflate the two, the better.

I hope I can convince you 😄